src/lib/navigation/base-navigation-mode-factory.provider.ts

```diff
@@ -1,72 +0,0 @@
-import {FreeNavigationMode} from './free-navigation-mode';
-import {NavigationMode} from './navigation-mode.interface';
-import {SemiStrictNavigationMode} from './semi-strict-navigation-mode';
-import {StrictNavigationMode} from './strict-navigation-mode';
-import {WizardComponent} from '../components/wizard.component';
-import {NavigationModeInput} from './navigation-mode-input.interface';
-import {NavigationModeFactory} from './navigation-mode-factory.interface';
-
-/**
- * A factory used to create [[NavigationMode]] instances
- */
-export class BaseNavigationModeFactory implements NavigationModeFactory {
-
-  /**
-   * @inheritDoc
-   */
-  public create(wizard: WizardComponent, navigationModeInput: NavigationModeInput): NavigationMode {
-    if (typeof navigationModeInput === 'string') {
-      // input is a name
-      return this.createByName(wizard, navigationModeInput);
-    } else if (navigationModeInput) {
-      // input is a NavigationMode instance
-      return navigationModeInput;
-    } else {
-      // use default
-      return this.createDefault(wizard);
-    }
-  }
-
-  /**
-   * Create a [[NavigationMode]] for the given wizard instance by a navigation mode name
-   *
-   * @param wizard The wizard componenent where the created [[NavigationMode]] will be used
-   * @param navigationModeInput The name of a built-in navigation mode or a custom navigation mode
-   * @returns The created [[NavigationMode]]
-   */
-  protected createByName(wizard: WizardComponent, navigationModeInput: string): NavigationMode {
-    switch (navigationModeInput) {
-      case 'free':
-        return new FreeNavigationMode();
-      case 'semi-strict':
-        return new SemiStrictNavigationMode();
-      case 'strict':
-        return new StrictNavigationMode();
-      default:
-        return !navigationModeInput ? this.createDefault(wizard) : this.createUnknown(wizard, navigationModeInput);
-    }
-  }
-
-  /**
-   * Create a [[NavigationMode]] for the given wizard instance which does not have a configured navigation mode
-   *
-   * @param wizard The wizard componenent where the created [[NavigationMode]] will be used
-   * @returns The created [[NavigationMode]]
-   */
-  protected createDefault(wizard: WizardComponent): NavigationMode {
-    return new StrictNavigationMode();
-  }
-
-  /**
-   * Create a [[NavigationMode]] for the given wizard instance by a not recognized navigation mode name
-   *
-   * The base implementation always throws an Error.
-   *
-   * @param wizard The wizard componenent where the created [[NavigationMode]] will be used
-   * @param navigationModeInput The name of a custom navigation mode
-   * @returns The created [[NavigationMode]]
-   */
-  protected createUnknown(wizard: WizardComponent, navigationModeInput: string): NavigationMode {
-    throw new Error(`Unknown navigation mode name: ${navigationModeInput}`);
-  }
-}
```

Comment:
angular-archwizard no longer recognizes navigation mode names (`strict`, `semi-strict`, `free`), which makes `NavigationModeFactory` classes redundant.

Reply:
Do we need to offer backward compatibility by still supporting `<aw-wizard navigationMode=strict|semi-strict|free>`? I do not think so, because the migration is quite simple

`navigationMode="strict"` - just remove, because this navigation mode is default
`navigationMode="semi-strict"` - `[awNavigationMode] navigateForward="allow"`
`navigationMode="free"` - `[awNavigationMode] navigateForward="allow"`
(see [discussion ](https://github.com/madoar/angular-archwizard/pull/211#discussion_r283084060) about why `"semi-strict"` and `"free"` become equal)

Reply:
I also don't think we need to offer backwards compatibility in this case.
The reason for me is that we already included a massive breaking change in the next version (#200). Therefore it should be ok to break it a bit more.